### PR TITLE
Feature: Support CreateURem

### PR DESCRIPTION
**Pull request by ovr**
Hey!

Adds the `IRBuilder` method [`CreateURem`](http://llvm.org/doxygen/classllvm_1_1IRBuilder.html#a4ef70bab263e38c5e0b8c1bf95a5d814).

Thanks

### CODE REVIEW

**Comment by ovr** [comment]
@MichaReiser Random crash on CI? Weird

File: src/ir/ir-builder.h [hunk 1]
```diff
@@ -31,29 +31,56 @@ class IRBuilderWrapper: public Nan::ObjectWrap, public FromValueMixin<IRBuilderW
     template<ConvertOperationFn method>
     static NAN_METHOD(ConvertOperation);
 
+    static NAN_METHOD(CreateAdd);
```
Comment (ovr) [inline src/ir/ir-builder.h]: @MichaReiser 
image

https://github.com/nodejs/nan/blob/master/doc/methods.md#method-declaration

Should we drop it ( NAN v1 method declarations.)?

Comment (MichaReiser) [reply src/ir/ir-builder.h]: Thanks for the input. Wasn't aware that the Macro isn't longer the suggested way. Let us look at this in a separate issue/PR.

File: src/ir/ir-builder.h [hunk 2]
```diff
@@ -31,29 +31,56 @@ class IRBuilderWrapper: public Nan::ObjectWrap, public FromValueMixin<IRBuilderW
...
     static NAN_METHOD(CreateFNeg);
     static NAN_METHOD(CreateGlobalStringPtr);
     static NAN_METHOD(CreateGlobalString);
-    static NAN_METHOD(CreateInBoundsGEP);
     static NAN_METHOD(CreateInBoundsGEPWithoutType);
     static NAN_METHOD(CreateInBoundsGEPWithType);
-    static NAN_METHOD(CreateInsertValue);
+    static NAN_METHOD(CreateInBoundsGEP);
     static NAN_METHOD(CreateIntCast);
+    static NAN_METHOD(CreateInsertValue);
```
Comment (MichaReiser) [inline src/ir/ir-builder.h]: What's the reason you declared all these methods on IRBuilder? The header only specifies functions that are defined in the implementation file. However, the C++ implementation does not define all these methods. These methods are all implemented by the template `NANBinaryOperation` and only exposed to JavaScript (you can't call these methods on the C++ implementation).

```suggestion
```

File: src/ir/ir-builder.h [hunk 3]
```diff
@@ -31,29 +31,56 @@ class IRBuilderWrapper: public Nan::ObjectWrap, public FromValueMixin<IRBuilderW
...
+    static NAN_METHOD(CreateICmpEQ);
```
Comment (MichaReiser) [inline src/ir/ir-builder.h]:
```suggestion
```

File: src/ir/ir-builder.h [hunk 4]
```diff
@@ -31,29 +31,56 @@ class IRBuilderWrapper: public Nan::ObjectWrap, public FromValueMixin<IRBuilderW
...
+    static NAN_METHOD(CreateICmpNE);
```
Comment (MichaReiser) [inline src/ir/ir-builder.h]:
```suggestion
```

**Pull request merged**